cloud/blockstore/libs/storage/partition_nonrepl/part_nonrepl_actor_checksumblocks.cpp

```diff
@@ -273,20 +178,25 @@ void TDiskAgentChecksumActor::HandleChecksumDeviceBlocksResponse(
     Done(ctx, std::move(response), EStatus::Success);
 }
 
-STFUNC(TDiskAgentChecksumActor::StateWork)
+bool TDiskAgentChecksumActor::OnMessage(TAutoPtr<NActors::IEventHandle>& ev)
 {
-    TRequestScope timer(*RequestInfo);
-
     switch (ev->GetTypeRewrite()) {
-        HFunc(TEvents::TEvWakeup, HandleTimeout);
-
-        HFunc(TEvDiskAgent::TEvChecksumDeviceBlocksRequest, HandleChecksumDeviceBlocksUndelivery);
-        HFunc(TEvDiskAgent::TEvChecksumDeviceBlocksResponse, HandleChecksumDeviceBlocksResponse);
-
-        default:
-            HandleUnexpectedEvent(ev, TBlockStoreComponents::PARTITION_WORKER);
-            break;
+        case TEvDiskAgent::TEvChecksumDeviceBlocksRequest::EventType: {
```

Comment:
maybe you can use `HFunc` here and in other places to simplify the code

Reply:
I need to know if the message has been processed or not.  Hfunc does not provide such an opportunity

Reply:
something like this perhaps ? 
```c++
bool TDiskAgentChecksumActor::OnMessage(TAutoPtr<NActors::IEventHandle>& ev)
{
    switch (ev->GetTypeRewrite()) {
        HFunc(TEvDiskAgent::TEvChecksumDeviceBlocksRequest, HandleChecksumDeviceBlocksUndelivery);
        HFunc(TEvDiskAgent::TEvChecksumDeviceBlocksResponse, HandleChecksumDeviceBlocksResponse);
        default:
            return false
    }
    return true;
}
```